pkg/crdpuller/discovery.go

```diff
@@ -105,8 +106,23 @@ func (sp *schemaPuller) PullCRDs(context context.Context, resourceNames ...strin
 		return nil, err
 	}
 
+	groupResources, err := restmapper.GetAPIGroupResources(sp.discoveryClient)
+	if err != nil {
+		return nil, err
+	}
+	mapper := restmapper.NewDiscoveryRESTMapper(groupResources)
```

Comment:
Why not `restmapper.NewDeferredDiscoveryRESTMapper` so we don't make the entire discovery data pull part of the critical path here?

Reply:
SGTM

Reply:
done 👍

Reply:
Although, I am not sure it will be effective. Few lines after there is a call for ServerPreferredResources which triggers lot of api calls.

Reply:
This is run every minute. Good enough for now.

Reply:
+1 that this optimization does not help.